authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AuthorizationResourceAction.java

```diff
@@ -93,4 +99,9 @@ public String getRight() {
     public AuthorizationResourceType getAuthorizationResourceType() {
         return authorizationResourceType;
     }
+
+    public static AuthorizationResourceAction getByRight(String right) {
+        return Arrays.stream(AuthorizationResourceAction.values()).filter(action -> StringUtils.equals(action.getRight(), right))
+                .findFirst().orElseThrow(() -> new NotFoundException(String.format("Action not found by right %s", right)));
+    }
```

Comment:
Can you please use a constant map instead to be more performant?

```java
    private static final Map<String, AuthorizationResourceAction> BY_NAME = Stream.of(AuthorizationResourceAction.values())
            .collect(Collectors.toUnmodifiableMap(AuthorizationResourceAction::getRight, Function.identity()));

    public static AuthorizationResourceAction getByRight(String right) {
        AuthorizationResourceAction result = BY_NAME.get(right);
        if (result == null) {
            throw new NotFoundException(String.format("Action not found by right %s", right));
        }
        return result;
    }
```

Reply:
it wont work, right field can be the same for more enum values, thus it will throw duplicatekeyexception

Reply:
fixed